File: backend/app/schemas/survey.py
from datetime import datetime
from pydantic import BaseModel


class QuestionInput(BaseModel):
    text: str
    question_type: str = "text"


class CreateSurveyRequest(BaseModel):
    title: str
    description: str = ""
    course_id: str | None = None
    questions: list[QuestionInput] = []


class SubmitResponseRequest(BaseModel):
    question_id: str
    respondent_name: str = "Anonymous"
    answer: str
    satisfaction: int | None = None


class QuestionResponse(BaseModel):
    id: str
    text: str
    question_type: str
    order_index: int

    model_config = {"from_attributes": True}


class FeedbackResponse(BaseModel):
    id: str
    question_id: str
    respondent_name: str
    answer: str
    satisfaction: int | None

    model_config = {"from_attributes": True}


class SurveyDetailResponse(BaseModel):
    id: str
    title: str
    description: str
    status: str
    created_at: datetime
    questions: list[QuestionResponse] = []
    responses: list[FeedbackResponse] = []

    model_config = {"from_attributes": True}


class GenerateQuestionsRequest(BaseModel):
    course_title: str
    course_style: str = ""
    course_theme: str = ""


class GenerateReplyRequest(BaseModel):
    feedback: str
    question: str

File: frontend/src/types/survey.ts
export interface SurveyQuestion {
  id?: string;
  text: string;
  question_type: string;
  order_index?: number;
}

export interface SurveyFeedback {
  id: string;
  question_id: string;
  respondent_name: string;
  answer: string;
  satisfaction: number | null;
}

export interface Survey {
  id: string;
  title: string;
  description: string;
  status: string;
  created_at: string;
  questions: SurveyQuestion[];
  responses: SurveyFeedback[];
}

export interface CreateSurveyRequest {
  title: string;
  description?: string;
  course_id?: string;
  questions: { text: string; question_type?: string }[];
}

export interface GenerateQuestionsRequest {
  course_title: string;
  course_style?: string;
  course_theme?: string;
}

export interface GenerateReplyRequest {
  feedback: string;
  question: string;
}
